### PR TITLE
Scala reflection now supports Java CRTP

### PR DESCRIPTION
**Pull request by xeno-by**

### CODE REVIEW

**Comment by xeno-by** [comment]
review @paulp @retronym

File: src/partest/scala/tools/partest/JarTest.scala [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+ */
+abstract class JarTest extends ReplTest {
+  override def transformSettings(settings: Settings): Settings = {
+    val thisFile = testPath.jfile.getAbsolutePath
+    val jarFile = (thisFile stripSuffix "scala") + "jar"
+    val classpath = List(sys.props("partest.lib"), sys.props("partest.reflect"), sys.props("partest.comp"), jarFile) mkString sys.props("path.separator")
+    settings.processArguments(List("-cp", classpath), true)
+    settings
+  }
+}
```
Comment (paulp) [inline src/partest/scala/tools/partest/JarTest.scala]: I'm glad to see the reuse effort. But, I see no reason we shouldn't handle the jar universally, just as we do with .flags and .check files.  Also, this trait is documented to be about putting a single jar on the classpath, but then it proceeds to place the scala library/reflect/compiler jars on as well.  Finally, we have way too much partest behavior hidden in system properties already, which is a complete PITA when trying to reproduce a partest result from outside of partest.  I don't see that we need separate system properties for all those jars.  How about we just pass a classpath.

**Comment by xeno-by** [comment]
PLS REBUILD ALL

**Comment by gkossakowski** [comment]
What's Java CRTP?

**Comment by soc** [comment]
CRTP == Curiously Recurring Template Pattern

In the enum case: http://stackoverflow.com/questions/648188/explain-the-generics-used-in-the-enum-declaration
(Sorry, too lazy to google for the right JavaDoc ...)

**Comment by xeno-by** [comment]
lgtm anyone?

**Comment by jsuereth** [comment]
LGTM

**Pull request merged**